src/home/celebration/NftReward.tsx

```diff
@@ -0,0 +1,245 @@
+  const expirationDate = new Date(nftCelebration?.expirationDate ?? 0)
+  const reminderDate = new Date(nftCelebration?.reminderDate ?? 0)
+  const deepLink = nftCelebration?.deepLink ?? ''
+
+  const { expirationStatus, expirationLabelText } = useExpirationStatus(
+    expirationDate,
+    reminderDate
+  )
+
+  const isVisible = canShowNftReward && matchingNft
+
+  useEffect(() => {
+    if (isVisible) {
+      // Wait for the home screen to have less ongoing async tasks.
+      // This should help the bottom sheet animation run smoothly.
+      const timeoutId = setTimeout(() => {
+        bottomSheetRef.current?.expand()
+      }, 1000)
+      return () => clearTimeout(timeoutId)
+    }
+  }, [isVisible])
+
+  const handleBottomSheetPositionChange = (index: number) => {
+    if (!celebratedNft) {
+      return // This should never happen
+    }
+
+    if (index === -1) {
+      ValoraAnalytics.track(HomeEvents.nft_reward_dismiss, {
+        networkId: celebratedNft.networkId,
+        contractAddress: celebratedNft.contractAddress,
+        remainingDays: differenceInDays(expirationDate, Date.now()),
+      })
+
+      dispatch(nftRewardDisplayed())
+    }
+  }
+
+  const handleCtaPress = () => {
+    if (!celebratedNft) {
+      return // This should never happen
+    }
+
+    ValoraAnalytics.track(HomeEvents.nft_reward_accept, {
+      networkId: celebratedNft.networkId,
+      contractAddress: celebratedNft.contractAddress,
+      remainingDays: differenceInDays(expirationDate, Date.now()),
+    })
+
+    bottomSheetRef.current?.close()
+
+    const isSecureOrigin = true
+    dispatch(openDeepLink(deepLink, isSecureOrigin))
+  }
+
+  const pillLabel =
+    expirationStatus === ExpirationStatus.expired
+      ? t('nftCelebration.rewardBottomSheet.expired')
+      : t('nftCelebration.rewardBottomSheet.expirationLabel', { expirationLabelText })
+
+  if (!isVisible) {
+    return null
+  }
+
+  return (
+    <BottomSheetBase forwardedRef={bottomSheetRef} onChange={handleBottomSheetPositionChange}>
+      <BottomSheetView style={[styles.container, insetsStyle]}>
+        <ExpirationPill status={expirationStatus} label={pillLabel} />
+        <Text style={styles.title}>{t('nftCelebration.rewardBottomSheet.title')}</Text>
+        <Text style={styles.desctiption}>
+          {t('nftCelebration.rewardBottomSheet.description', {
+            nftName: matchingNft.metadata.name,
+          })}
+        </Text>
+        <Button
+          style={styles.button}
+          type={BtnTypes.PRIMARY}
+          size={BtnSizes.FULL}
+          disabled={expirationStatus === ExpirationStatus.expired}
+          onPress={handleCtaPress}
+          text={t('nftCelebration.rewardBottomSheet.cta')}
+        />
+      </BottomSheetView>
+    </BottomSheetBase>
+  )
+}
+
+enum ExpirationStatus {
+  active = 'active',
+  aboutToExpire = 'aboutToExpire',
+  expired = 'expired',
+}
+
+// recalculates expiration status every second
```

Comment:
if i understand correctly, this is necessary only towards the end of the reward period? i think that we can simplify this by setting an expiry date in statsig that is like 30min-1 hour before the actual expiry date - the worst case is that a user who is seeing this really close to the end of the program does not see the reward bottom sheet, however i think this tradeoff is worth the reduced code complexity? (we can also think of it like, we allow a grace period in the backend from the expiry time)

Reply:
yes, it's mostly for handling the situation when the reward expires while the bottom sheet is open.

I agree that we could expect that user wouldn't spend too much time looking at this bottom sheet and it's acceptable to not update expiration time and not auto dismiss the bottom sheet once it expires.

I like the idea of having a grace period, giving the late comers the chance to exercise their rewards.